packages/core/src/browser/xhrProxy.ts

```diff
@@ -69,85 +80,84 @@ export function resetXhrProxy() {
   }
 }
 
-function proxyXhr() {
-  // eslint-disable-next-line @typescript-eslint/unbound-method
-  originalXhrOpen = XMLHttpRequest.prototype.open
-  // eslint-disable-next-line @typescript-eslint/unbound-method
-  originalXhrSend = XMLHttpRequest.prototype.send
-  // eslint-disable-next-line @typescript-eslint/unbound-method
-  originalXhrAbort = XMLHttpRequest.prototype.abort
-
-  XMLHttpRequest.prototype.open = function (this: BrowserXHR, method: string, url: string) {
-    callMonitored(() => {
-      // WARN: since this data structure is tied to the instance, it is shared by both logs and rum
-      // and can be used by different code versions depending on customer setup
-      // so it should stay compatible with older versions
-      ;(this._datadog_xhr as Pick<XhrStartContext, 'method' | 'url'>) = {
-        method,
-        url: normalizeUrl(url),
-      }
-    })
-    return originalXhrOpen.apply(this, arguments as any)
-  }
+function openXhr(this: BrowserXHR<XhrOpenContext>, method: string, url: string) {
+  callMonitored(() => {
+    // WARN: since this data structure is tied to the instance, it is shared by both logs and rum
+    // and can be used by different code versions depending on customer setup
+    // so it should stay compatible with older versions
+    this._datadog_xhr = {
+      method,
+      url: normalizeUrl(url),
+    }
+  })
+  return originalXhrOpen.apply(this, arguments as any)
+}
+
+function sendXhr(this: BrowserXHR) {
+  callMonitored(() => {
+    if (!this._datadog_xhr) {
+      return
+    }
+    this._datadog_xhr = {
+      ...this._datadog_xhr,
+      startTime: relativeNow(),
+      startClocks: clocksNow(),
+      isAborted: false,
+    }
```

Comment:
Since it is available publicly, I'd be in favor of keeping the actual behavior which keep the same reference for this object during the lifecycle of the xhr.

Reply:
ok